File: Leetcode/2078/2078.cpp
class Solution {
public:
    int maxDistance(vector<int>& colors) {
        int n = (int)colors.size();
        int ans = 0;
        for (int i = 0; i < n-1; ++i) {
            for (int j = i+1; j < n; ++j) {
                if (colors[i] != colors[j]) {
                    ans = max(ans, abs(i-j));
                }
            }
        }
        return ans;
    }
};

File: Leetcode/2078/2078.go
func maxDistance(colors []int) int {
	ans := 0
	var temp int
	for i := 0; i < len(colors)-1; i++ {
		for j := i+1; j < len(colors); j++ {
			if colors[i] != colors[j] {
				temp = j-i
				if temp > ans {
					ans = temp
				}
			}
		}
	}
	return ans
}